modules/apps/roll-dialog/cast-dialog.js

```diff
@@ -31,6 +31,7 @@ export default class CastDialog extends SkillDialog {
 
         data.skill = spell.skillToUse;
         data.characteristic = data.skill?.system?.characteristic?.key || "int";
+        fields.unofficialGrimoire = game.settings.get("wfrp4e", "unofficialgrimoire");
```

Comment:
Fields are generally reserved for properties that are actually part of the dialog, visible to the user. This should go in `data`